tests/CppUTestExt/MockSupportTest.cpp

```diff
@@ -233,3 +233,19 @@ TEST(MockSupportTestWithFixture, ShouldNotCrashOnFailureAfterCrashMethodWasReset
     CHECK_FALSE(cpputestHasCrashed);
 }
 
+TEST(MockSupportTestWithFixture, ShouldFailWhenRepeated)
+{
+    fixture.setTestFunction(unexpectedTestFunction_);
+
+    fixture.runAllTests();
+    fixture.assertPrintContains("Unexpected call to function: unexpected");
+
+    delete fixture.result_;
+    delete fixture.output_;
+    fixture.output_ = new StringBufferTestOutput();
+    fixture.result_ = new TestResult(*fixture.output_);
+
```

Comment:
The reason I am not using a "fresh" fixture is that I want the registry to remain the same, as is the case when a real repeat is done.